test/unit/style-spec/fixture/sources.output.json

```diff
@@ -7,14 +7,6 @@
     "message": "sources.invalid-type.type: expected one of [vector, raster, raster-dem, geojson, video, image], \"invalid\" found",
     "line": 7
   },
-  {
-    "message": "sources.no-tilejson-properties-with-url.tiles: a source with a \"url\" property may not include a \"tiles\" property",
-    "line": 12
-  },
-  {
-    "message": "sources.no-unknown-properties-with-url.foo: a source with a \"url\" property may not include a \"foo\" property",
```

Comment:
Why is unknown properties support needed?

Reply:
I think foreign members (unknown properties) in source options should be leaved as it is. Foreign members does not affect source's functionality and add extensibility for metadata or custom sources. As we don't invalidate foreign members in geojson、image、video sources, invalidate foreign members in  vector, raster, raster-dem seems inconsistent.

Reply:
@jingsam Sorry for the delay in getting back to this. Allowing foreign members in the TileJSON does not seem necessary as part of this PR. It is preferable to have strict validation to ensure that unnecesary or incorrect properties are not being added to such files.